opentracing-api/src/main/java/io/opentracing/ActiveSpan.java

```diff
@@ -0,0 +1,71 @@
+ * {@link ActiveSpanSource#active()} method. If the application needs to defer work that should be part of the same Span, the
+ * Source provides a {@link ActiveSpan#defer} method that returns a {@link Continuation}; this continuation may be used
+ * to re-activate and continue the {@link Span} in that other asynchronous executor and/or thread.
+ *
+ * <p>
+ * {@link ActiveSpan}s are created via {@link Tracer.SpanBuilder#startAndActivate()} or {@link ActiveSpanSource#adopt}. They can
+ * be {@link ActiveSpan#defer()}ed as {@link ActiveSpan.Continuation}s, then re-{@link Continuation#activate()}d later.
+ *
+ * @see ActiveSpanSource
+ */
+public interface ActiveSpan extends Closeable, Span {
+    /**
+     * Mark the end of the active period for the {@link Span} pinned by this {@link ActiveSpan}. When the last
+     * {@link ActiveSpan} is deactivated for a given {@link Span}, it is automatically {@link Span#finish()}ed.
+     * <p>
+     * <p>
+     * NOTE: It is an error to call deactivate() more than once on a single {@link ActiveSpan}.
+     *
+     * @see Closeable#close() {@link ActiveSpan}s are auto-closeable and may be used in try-with-resources blocks
+     */
+    void deactivate();
+
+    /**
+     * "Fork" a new {@link Continuation} associated with this {@link ActiveSpan} and {@link Span}, as well as any
+     * 3rd-party execution context of interest.
+     * <p>
+     * <p>
+     * The associated {@link Span} will not {@link Span#finish()} while a {@link Continuation} is outstanding; in
+     * this way, it provides a reference/pin just like an active @{Handle} does.
+     *
+     * @return a new {@link Continuation} to {@link Continuation#activate()} at the appropriate time.
+     */
+    Continuation defer();
+
+    /**
+     * A {@link Continuation} can be used *once* to activate a Span along with any non-OpenTracing execution context
+     * (e.g., MDC), then deactivate when processing activity moves on to another Span. (In practice, this active period
+     * typically extends for the length of a deferred async closure invocation.)
+     *
+     * <p>
+     * Most users do not directly interact with {@link Continuation}, {@link Continuation#activate()} or
+     * {@link ActiveSpan#deactivate()}, but rather use {@link ActiveSpanSource}-aware Runnables/Callables/Executors.
+     * Those higher-level primitives need not be defined within the OpenTracing core API, and so they are not.
+     *
+     * <p>
+     * NOTE: {@link Continuation} extends {@link Closeable} rather than AutoCloseable in order to keep support
+     * for JDK1.6.
+     *
+     * @see ActiveSpanSource#adopt(Span)
+     */
+    interface Continuation {
+        /**
+         * Make the Span (and other execution context) encapsulated by this Continuation active and return it.
+         *
+         * <p>
+         * NOTE: It is an error to call activate() more than once on a single Continuation instance.
```

Comment:
@bhs there are some downsides to NoSpan concept that were discussed in last year PRs about whether extract() should return null or not. One of them is potential compatibility issue with existing instrumentation that checks for null. 

If we want to introduce NoSpan, then I would consider merging it with the no-op tracer implementation, rather than a standalone thing. And I would rather have it as a public immutable singleton, rather than a marker interface, so that it can be compared  via `==`.

Agree that it should go as a separate PR.